src/argilla/client/feedback/dataset/local/dataset.py

```diff
@@ -247,6 +259,26 @@ def add_metadata_property(
         self._metadata_properties.append(metadata_property)
         return metadata_property
 
+    def vector_settings_by_name(self, name: str) -> VectorSettings:
+        vector_settings = self._vector_settings.get(name)
+        if not vector_settings:
+            raise KeyError(f"Vector settings with name '{name!r}' does not exist in the dataset.")
+
+        return vector_settings
+
+    def add_vector_settings(self, vector_settings: VectorSettings) -> VectorSettings:
+        if self._vector_settings.get(vector_settings.name):
+            raise ValueError(f"Vector settings with name '{vector_settings.name}' already exists in the dataset.")
+
+        self._vector_settings[vector_settings.name] = vector_settings
+        return vector_settings
```

Comment:
Something similar here. Do we want to raise an exception or silently overwrite the specified vector settings instead?
```suggestion
    def add_vector_settings(self, vector_settings: VectorSettings) -> VectorSettings:
        self._vector_settings[vector_settings.name] = vector_settings
        
        return vector_settings
```

Reply:
Again, this is aligned with other `add_*` methods. Anyway, in this case could make sense to warn at least the user since there is already a previous vector setting, and can be useful to users to detect wrong configuration definitions.